api/notd/api/gallery_v1.py

```diff
@@ -13,4 +17,15 @@ async def list_collection_token_airdrops(registryAddress: str, tokenId: str):
         airdrops = await galleryManager.list_collection_token_airdrops(registryAddress=registryAddress, tokenId=tokenId)
         return ListCollectionTokenAirdropsResponse(airdrops=(await responseBuilder.airdrops_from_models(airdrops=airdrops)))
 
+    @router.get('/collections/{registryAddress}/attributes', response_model=GetCollectionAttributesResponse)
+    async def get_collection_attributes(registryAddress: str):
+        attributes = await galleryManager.get_collection_attributes(registryAddress=registryAddress)
+        return GetCollectionAttributesResponse(attributes=(await responseBuilder.attributes_from_models(attributes=attributes)))
+
+    @router.get('/collections/{registryAddress}/tokens', response_model=GetTokensWithAttributeResponse)
+    async def get_tokens_with_attributes(registryAddress: str, attributeName: Optional[str], attributeValue: Optional[str],attributeName2: Optional[str], attributeValue2: Optional[str], limit: Optional[int] = None, offset: Optional[int] = None):
```

Comment:
attributeName and attributeName2 are just examples. really it would be like `?hair=yello&teeth=orange`